### PR TITLE
refactor: get default network from advanced logic

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

**Review by benjlevesque — COMMENTED** [review]
I find this change surprising. It feels we should actually _remove_ the default value, and ensure a network is always specified

**Review by benjlevesque — APPROVED** [review]

File: packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import Utils from '@requestnetwork/utils';
 import ReferenceBased from './reference-based';
 
 const CURRENT_VERSION = '0.1.0';
+// Default network if nothing is given in the data
```
Comment (yomarion) [inline packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts:6]:
```suggestion
// Default network if the storage data does not give any
```

File: packages/payment-processor/src/payment/conversion-erc20.ts [hunk 2]
```diff
@@ -26,8 +26,10 @@ export async function approveErc20ForProxyConversionIfNeeded(
   overrides?: ITransactionOverrides,
 ): Promise<ContractTransaction | void> {
   const network =
-    request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY].values.network ||
-    'mainnet';
+    request.extensions[ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY].values.network;
+  if (!network) {
+    throw new Error(`Payment network currency must have a network`);
+  }
```
Comment (yomarion) [inline packages/payment-processor/src/payment/conversion-erc20.ts:32]: Can't this be handled in the type definition of extensions ? Now that we have a default value, it should be useless to test the presence of this value at higher level methods.

File: packages/payment-processor/src/payment/swap-any-to-erc20.ts [hunk 3]
```diff
@@ -31,9 +30,12 @@ export async function swapToPayAnyToErc20Request(
...
     throw new Error(`The request must have the payment network any-to-erc20-proxy`);
   }
+
   const network =
-    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network ||
-    'mainnet';
+    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network;
+  if (!network) {
+    throw new Error(`Payment network currency must have a network`);
+  }
```
Comment (yomarion) [inline packages/payment-processor/src/payment/swap-any-to-erc20.ts:38]: Same remark, can't we keep these kinds of checks at the lower levels?

File: packages/payment-processor/src/payment/swap-any-to-erc20.ts [hunk 4]
```diff
@@ -68,7 +70,10 @@ export function encodeSwapToPayAnyToErc20Request(
   if (!swapSettings) {
     throw new Error(`Swap Settings are required`);
   }
-  const network = conversionSettings.currency?.network || 'mainnet';
+  const network = conversionSettings.currency?.network;
+  if (!network) {
+    throw new Error(`Currency in conversion settings must have a network`);
+  }
```
Comment (yomarion) [inline packages/payment-processor/src/payment/swap-any-to-erc20.ts:76]: Same remark, can't we keep these kinds of checks at the lower levels?

File: packages/payment-processor/src/payment/swap-conversion-erc20.ts [hunk 5]
```diff
@@ -62,8 +62,10 @@ export async function approveErc20ForSwapWithConversionToPay(
   overrides?: ITransactionOverrides,
 ): Promise<ContractTransaction> {
   const network =
-    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network ||
-    'mainnet';
+    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network;
+  if (!network) {
+    throw new Error(`Payment network currency must have a network`);
+  }
```
Comment (yomarion) [inline packages/payment-processor/src/payment/swap-conversion-erc20.ts:68]: Same remark, can't we keep these kinds of checks at the lower levels?

**Review by yomarion — COMMENTED** [review]
I think if we go one step further and enforce the network definition in the type, it will enlight the code and improve maintenability, while decreasing the need for tests.

**Pull request merged**